### PR TITLE
Alerting: Read group details before saving

### PR DESCRIPTION
**Pull request by gillesdemey**
**What this PR does / why we need it**:

Allows alert groups to be edited in separate tabs or by multiple editors while preserving changes that have been made by other authors by reading the state of the alert group before mutating.

**Special notes for your reviewer**:

Added a safe-guard to `addRulerGroupsToCombinedNamespace` unrelated to the above.

### CODE REVIEW

File: public/app/features/alerting/unified/utils/rulerClient.ts [hunk 1]
```diff
@@ -157,16 +157,22 @@ export function getRulerClient(rulerConfig: RulerDataSourceConfig): RulerClient
       return addRuleToNamespaceAndGroup(namespace, groupSpec, newRule);
     }
 
-    const sameNamespace = existingRule.namespace === namespace;
-    const sameGroup = existingRule.group.name === values.group;
+    // we'll fetch the existing group again, someone might have updated it while we were editing a rule
+    const freshExisting = await findEditableRule(ruleId.fromRuleWithLocation(existingRule));
+    if (!freshExisting) {
+      throw new Error('Rule not found.');
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/utils/rulerClient.ts:163]: When is this possible??

Comment (gillesdemey) [reply public/app/features/alerting/unified/utils/rulerClient.ts:163]: It shouldn't really – I tried to replicate this scenario but couldn't. Regardless it's typed as such and needs to be handled :D

**Review by konrad147 — APPROVED** [review]
LGTM!

File: public/app/features/alerting/unified/utils/rulerClient.ts (hunk 1 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/utils/rulerClient.ts:163]: 💯, I was only curious

**Review by soniaAguilarPeiron — APPROVED** [review]

**Pull request merged**